src/query/service/src/servers/http/clickhouse_handler.rs

```diff
@@ -112,7 +112,14 @@ async fn execute(
     handle: Option<JoinHandle<()>>,
 ) -> Result<WithContentType<Body>> {
     let format_typ = format.typ.clone();
-    let mut data_stream = interpreter.execute(ctx.clone()).await?;
+    let mut data_stream = ctx
+        .try_spawn({
+            let ctx = ctx.clone();
+            async move { interpreter.execute(ctx.clone()).await }
+        })?
+        .await
+        .map_err(ErrorCode::from_std_error)??;
```

Comment:
`from_std_error` will lost error context, how about using `map_err(|err| anyhow!("clickhouse handler failed to join interpreter thread: {err:?}"))` for better error message?

Reply:
yeah, that will be better

Reply:
PTAL, the error message has been updated as you suggested